### PR TITLE
 bigquery: allow schema-ful data listing

### PR DESCRIPTION
**Pull request by pongad**
listTableData and query now use the same classes.

### CODE REVIEW

**Comment by tswast** [comment]
Looks like it's missing an update to the integration tests.

```
[ERROR] /home/travis/build/GoogleCloudPlatform/google-cloud-java/google-cloud-examples/src/test/java/com/google/cloud/examples/bigquery/snippets/ITBigQuerySnippets.java:[39,33] cannot find symbol

[ERROR]   symbol:   class QueryResult
[ERROR]   location: package com.google.cloud.bigquery
```

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java [hunk 1]
```diff
@@ -445,46 +446,72 @@ public TableDataInsertAllResponse call() throws Exception {
...
+  public TableResult listTableData(
+      String datasetId, String tableId, TableDataListOption... options) {
+    return listTableData(TableId.of(datasetId, tableId), options);
   }
 
   @Override
-  public Page<FieldValueList> listTableData(TableId tableId, TableDataListOption... options) {
-    return listTableData(tableId, getOptions(), optionMap(options));
+  public TableResult listTableData(TableId tableId, TableDataListOption... options) {
+    return listTableData(tableId, null, options);
```
Comment (tswast) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java]: I'm having trouble following which overloaded method this is going to. If it's the one above, it seems to be using `tableId` as the `datasetId`.

I suppose it's intended that the Schema is null, but I'm not sure how Java knows that.

Also, what was BigQueryOptions for and why is it going away?

Comment (pongad) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java]: This is calling overload on line 466.
The first argument has type `TableId` which disqualifies L449 and L460, so L466 is the only one left with the right number of args.

The `BigQueryOptions` isn't going away, it's just being populated on L468.

I think this means we have too many overloads, but I'm not sure if we should require the user to pass null schema.

**Comment by pongad** [comment]
Build failure fixed now. Sorry, I keep missing them.

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java (hunk 1 above)
Comment (tswast) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java]: Thanks for clarifying. Yeah, too many overloads. But we shouldn't force them to explicitly pass in null when they don't know the schema.

**Pull request merged**